apisix/cli/ngx_tpl.lua

```diff
@@ -766,6 +766,7 @@ http {
             grpc_set_header   "Host" $upstream_host;
             {% end %}
             grpc_set_header   Content-Type application/grpc;
+            grpc_set_header   Te trailers;
```

Comment:
grpc-message: Missing :te header

This error is responsed by grpc server, not client. If not set this header, no request can received by grpc server.

see that :  https://github.com/grpc/grpc/blob/master/src/core/ext/filters/http/server/http_server_filter.cc#L93

Reply:
Not all grpc servers follow such rules, have you tested golang's grpc server?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/TE
TE can be set to multiple values, it should not hard-coded

Reply:
Thanks for disscussing.
grpc client hard-code set TE header.
golang grpc client:  https://github.com/grpc/grpc-go/blob/master/internal/transport/http2_client.go#L528
C/C++ grpc client:  https://github.com/grpc/grpc/blob/master/src/core/ext/filters/http/client/http_client_filter.cc#L118

grpc requires te header in their spec: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md, which is may different from your reference.

I believe that every grpc client should hard-code set te header.

So if you use golang grpc client to request python grpc server, it will be success. 
But you use Apisix to request python grpc server, it will fail.

Reply:
kong also hard-code te header in their grpc-gateway plugin.
https://github.com/Kong/kong/blob/master/kong/plugins/grpc-gateway/handler.lua#L54

Reply:
So we need to add this header to grpc-transcode plugin instead of ngx_tpl.lua?

Reply:
I find that headers are added in ngx_tpl.lua, such as  Content-Type. So I think Keep adding te header in ngx_tpl.lua is better.

Reply:
The `trailers` is needed in grpc.
1. status (grpc-status) is sent as a trailing header after the content is sent. 
2. support streaming use cases
So the request header `Te: trailers` is needed in grpc.

Reply:
@moonming The grpc-transcode plugin use the `location @grpc_pass` to pass request

Reply:
Any other suggession about this commit ?  It seems that this bug needs to be fixed,

Reply:
@moonming please help check this